### PR TITLE
Rhys/use hashed bundle

### PR DESCRIPTION
**Pull request by wheresrhys**
Also contains a load of (hopefully) simplification of how paths to assets are built.

### CODE REVIEW

File: build/app/download-assets.js [hunk 1]
```diff
@@ -5,45 +5,45 @@ const logger = require('./logger');
...
 							return res.text();
 						}
-						throw new Error('Failed to fetch n-ui stylesheet');
+						throw new Error('Failed to download ${s3Name} from s3');
 					})
 					.then(text => {
 						// if it's an empty string, something probably went wrong
 						if (!text.length) {
-							throw new Error('Fetched empty n-ui stylesheet');
+							throw new Error('Fetched empty ${s3Name} from s3');
```
Comment (kavanagh) [inline build/app/download-assets.js]: Backticks

File: build/app/download-assets.js [hunk 2]
```diff
@@ -5,45 +5,45 @@ const logger = require('./logger');
...
 		nUiManager.getReleaseRoots()
 			.map(urlRoot =>
-				nEagerFetch(`${urlRoot}head-n-ui-core.css`, {retry: 3})
+				nEagerFetch(`${urlRoot}${s3Name}`, {retry: 3})
 					.then(res => {
 						if (res.ok) {
 							return res.text();
 						}
-						throw new Error('Failed to fetch n-ui stylesheet');
+						throw new Error('Failed to download ${s3Name} from s3');
```
Comment (kavanagh) [inline build/app/download-assets.js]: Backticks

File: server/lib/asset-manager/asset-url-generator.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		return {};
+	}
+}
+
+module.exports = (appName, directory, useLocalAppShell) => {
+
+	const assetHashes = loadAssetHashesJson(`${directory}/public/asset-hashes.json`);
+	const nUiAssetHashes = loadAssetHashesJson(`${directory}/public/n-ui-asset-hashes.json`);
+	const nUiReleaseName = nUiManager.getReleaseName(directory);
+	const nUiUnhashedAssetsRoot = useLocalAppShell ? '/${appName}/n-ui/' : `//www.ft.com/__assets/n-ui/cached/${nUiReleaseName}/`;
```
Comment (kavanagh) [inline server/lib/asset-manager/asset-url-generator.js]: Backticks?

File: build/app/download-assets.js (hunk 2 above)
Comment (wheresrhys) [inline build/app/download-assets.js]: 🙀

File: server/lib/asset-manager/asset-url-generator.js (hunk 3 above)
Comment (wheresrhys) [inline server/lib/asset-manager/asset-url-generator.js]: should probably add some tests for dev mode. Not in this PR though as it's already epic

**Pull request merged**